packages/xrpl/test/models/oracleSet.test.ts

```diff
@@ -167,7 +167,14 @@ describe('OracleSet', function () {
   })
 
   it(`throws w/ invalid AssetPrice of PriceDataSeries`, function () {
-    tx.PriceDataSeries[0].PriceData.AssetPrice = '1234'
+    tx.PriceDataSeries[0].PriceData.AssetPrice = 'ghij' // value cannot be parsed as hexadecimal number
+    const errorMessage = 'OracleSet: invalid field AssetPrice'
+    assert.throws(() => validateOracleSet(tx), ValidationError, errorMessage)
+    assert.throws(() => validate(tx), ValidationError, errorMessage)
+  })
+
+  it(`throws w/ invalid AssetPrice type in PriceDataSeries`, function () {
+    tx.PriceDataSeries[0].PriceData.AssetPrice = ['sample', 'invalid', 'type']
```

Comment:
Can you also add a positive test case with AssetPrice using a valid string value?

Reply:
[35286b0](https://github.com/XRPLF/xrpl.js/pull/2913/commits/35286b03cf9c5e80e8622dfb532f59ce4f41204a)